packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -384,9 +384,9 @@ export default class {
 
     delete this._xapis[id]
 
-    const { pool } = xapi
-    if (pool != null) {
-      delete this._xapisByPool[pool.$id]
+    const xapisByPool = this._xapisByPool
+    for (const poolId in xapisByPool) {
```

Comment:
Because you're storing server's id and not `Xapi` objects, you can skip this step entirely and rely of the fact that there are no entries available in `this._xapis`.